src/backend/distributed/commands/multi_copy.c

```diff
@@ -2311,46 +2408,84 @@ CitusSendTupleToPlacements(TupleTableSlot *slot, CitusCopyDestReceiver *copyDest
 	/* connections hash is kept in memory context */
 	MemoryContextSwitchTo(copyDest->memoryContext);
 
-	/* get existing connections to the shard placements, if any */
-	shardConnections = GetShardHashConnections(shardConnectionHash, shardId,
-											   &shardConnectionsFound);
-	if (!shardConnectionsFound)
+	shardState = GetShardState(shardId, copyDest->shardStateHash,
```

Comment:
Minor: Instead of passing `found `, can't we use `shardState->placementStateList != NIL`? This is slightly hard to follow

Reply:
`shardState->placementStateList` is always initiated with the placement list when a `CopyShardState` created, so it is not NIL here.

I changed the names to make it a bit more readable.